src/org/zaproxy/zap/extension/ascanrulesBeta/SourceCodeDisclosureSVN.java

```diff
@@ -255,18 +273,14 @@ private boolean findSourceCodeSVN(HttpMessage originalMessage) throws Exception
 			//svnsourcefileattackmsg.setRequestHeader(this.getBaseMsg().getRequestHeader());
 			sendAndReceive(svnsourcefileattackmsg, false);  //do not follow redirects
 
-			//if we got a 404 or a redirect specifically, then this is NOT a match
-			//note that since we are simply relying on the file existing or not, we 
-			//will not attempt any fuzzy matching. Old school.
-			//this check is necessary, otherwise a recursive scan on nodes in the url path cause lots of false positives.
-			if (	svnsourcefileattackmsg.getResponseHeader().getStatusCode() !=  HttpStatusCode.NOT_FOUND &&  //404
-					svnsourcefileattackmsg.getResponseHeader().getStatusCode() !=  HttpStatusCode.MOVED_PERMANENTLY && //301
-					svnsourcefileattackmsg.getResponseHeader().getStatusCode() !=  HttpStatusCode.FOUND && //302
-					svnsourcefileattackmsg.getResponseHeader().getStatusCode() !=  HttpStatusCode.SEE_OTHER	&& //303
-					svnsourcefileattackmsg.getResponseHeader().getStatusCode() !=  HttpStatusCode.NOT_MODIFIED && //304
-					svnsourcefileattackmsg.getResponseHeader().getStatusCode() !=  HttpStatusCode.USE_PROXY && //305 (306 is currently unused)
-					svnsourcefileattackmsg.getResponseHeader().getStatusCode() !=  HttpStatusCode.TEMPORARY_REDIRECT //307
-					) {
+			int attackmsgResponseStatusCode = svnsourcefileattackmsg.getResponseHeader().getStatusCode();
+			//At LOW and MEDIUM ignore all client and server error responses
+			if ((attackStrength==AttackStrength.LOW || attackStrength==AttackStrength.MEDIUM) && 
+				(HttpStatusCode.isClientError(attackmsgResponseStatusCode) || HttpStatusCode.isServerError(attackmsgResponseStatusCode))) {
+					return false; 
+			}				
+			if (!unwantedCodes.contains(attackmsgResponseStatusCode)) {//If the response is wanted (not on the unwanted list)
+
```

Comment:
1) Yes I can switch it to Threshold, will just have to switch the logic, because the scales are inverted (Threshold vs Strength).
2) Yes your behavior observation is valid, at high strength you might get more false positives. (Or the other way around if we switch to Threshold at lower Threshold there might be more false positives because we relax the rules for raising an alert...so though we doubt a 504 is a valid result we'd still alert on it....)